### PR TITLE
add draft of 'pkgdb lock' subcommand

### PR DESCRIPTION
**Pull request by aakropotkin**
Adds a `pkgdb lock` subcommand which will lock registry/manifest inputs.

This iteration is functional but does not emit _valid_ `lockfile.json` format for all fields - simply locks the `registry` field for now.
Later PRs will handle other fields.

### CODE REVIEW

File: tests/registry.cc [hunk 1]
```diff
@@ -90,19 +90,19 @@ test_RegistryFileMixinGetRegWithoutFile()
...
+  /* Try loading the registry without setting the path. */
   try
     {
-      rfm.setRegistryPath( "" );
+      rfm.manifestPath = "";
       RegistryRaw regRaw = rfm.getRegistryRaw();
       return false;
     }
-  catch ( FloxException & )
+  catch ( ... )
```
Comment (aakropotkin) [inline tests/registry.cc:105]: TODO: Create a `FloxFileNotFound` exception

File: src/command.cc [hunk 2]
```diff
@@ -170,45 +170,91 @@ AttrPathMixin::fixupAttrPath()
...
+    return false;
+  };
+
+  bool have = this->manifestPath.has_value() ||
+              tryFile( "manifest.toml" ) ||
+              tryFile( "manifest.yaml" ) ||
+              tryFile( "manifest.json" ) ||
+              tryFile( ".flox/manifest.toml" ) ||
+              tryFile( ".flox/manifest.yaml" ) ||
+              tryFile( ".flox/manifest.json" );
```
Comment (mkenigs) [inline src/command.cc:214]: suggestion nonblocking: drop these and make `--manifest` required

File: src/command.cc [hunk 3]
```diff
@@ -170,45 +170,91 @@ AttrPathMixin::fixupAttrPath()
...
+      std::ostringstream oss;
+      oss << f.rdbuf();
+      this->contents = yamlToJSON( oss.str() );
+    }
+  else if ( ext == ".toml" )
     {
-      throw InvalidRegistryFileException( "registry path is null" );
+      std::ostringstream oss;
+      oss << f.rdbuf();
+      this->contents = tomlToJSON( oss.str() );
```
Comment (mkenigs) [inline src/command.cc:254]: suggestion nonblocking: only support TOML

**Review by mkenigs — APPROVED** [review]

File: src/command.cc (hunk 2 above)
Comment (aakropotkin) [reply src/command.cc:214]: Future proofeded

Comment (aakropotkin) [reply src/command.cc:214]: Y'all can delete the JSON and TOML options after doing user A/B testing and I swear I'll never utter the phrase "I told you so" 😶

File: src/pkgdb/pkg-query.cc [hunk 4]
```diff
@@ -66,9 +66,7 @@ PkgQueryArgs::InvalidPkgQueryArgException::errorMessage(
       case PQEC_INVALID_MATCH_STYLE:
         return "Query `matchStyle' must be set when `match' is used";
         break;
-      default:
-        return "Unexpected PkgQuery error";
-        break;
+      default: return "Unexpected PkgQuery error"; break;
```
Comment (bryanhonof) [inline src/pkgdb/pkg-query.cc:69]: style(non-blocking): Bothers me a bit that this is all on one line, compared to the above cases.

File: tests/registry.cc (hunk 1 above)
Comment (bryanhonof) [reply tests/registry.cc:105]: suggestion(non-blocking): Create an issue for it?

**Review by bryanhonof — APPROVED** [review]
lgtm.

File: src/pkgdb/pkg-query.cc (hunk 4 above)
Comment (aakropotkin) [reply src/pkgdb/pkg-query.cc:69]: I think the formatter did that, I don't remember editing that line.
I'll rerun it and see.

File: tests/registry.cc (hunk 1 above)
Comment (aakropotkin) [reply tests/registry.cc]: I went ahead and added error handling

File: src/command.cc (hunk 2 above)
Comment (aakropotkin) [reply src/command.cc]: I went ahead and changed it, and removed anything that tried to read from `PWD`

**Pull request merged**